src/smoosh/src/smoosh_channel.erl

```diff
@@ -194,21 +235,164 @@ handle_info(check_window, State0) ->
         end,
     erlang:send_after(60 * 1000, self(), check_window),
     {noreply, FinalState};
+handle_info(start_recovery, #state{name = Name, waiting = Waiting0} = State0) ->
+    RecActive = recover(active_file_name(Name)),
+    Waiting1 = lists:foldl(
+        fun(DbName, Acc) ->
+            case couch_db:is_compacting(DbName) of
+                true ->
+                    Priority = smoosh_server:get_priority(Name, DbName),
+                    smoosh_priority_queue:in(DbName, Priority, Priority, Acc);
+                false ->
+                    Acc
+            end
+        end,
+        Waiting0,
+        RecActive
+    ),
+    _ = State0#state{waiting = Waiting1},
+    ok = wait_for_active_restart(State0),
```

Comment:
We shouldn't block in gen_server callbacks. If we block here we would be waiting forever (until timeot). Since we never reach the call erlang:send_after(?ACTIVATE_DELAY_IN_MSEC. Even if we move send after we would still be blocked.

Reply:
I think we know the `waiting` is empty [here](https://github.com/noahshaw11/couchdb/blob/add-smoosh-queue-persistence/src/smoosh/src/smoosh_channel.erl#L425)

```
        length(State#state.active) + length(State#state.starting) < Concurrency ->
            case smoosh_priority_queue:out(State#state.waiting) of
                false ->
                    State;
```

Reply:
Yes, it seems like it from `smoosh_priority_queue:out/1`:
```
out(#priority_queue{name = Name, map = Map, tree = Tree}) ->
    case gb_trees:is_empty(Tree) of
        true ->
            false;
        false ->
            {_, {Key, Value}, Tree1} = gb_trees:take_largest(Tree),
            Map1 = maps:remove(Key, Map),
            Q = #priority_queue{name = Name, map = Map1, tree = Tree1},
            {Key, Value, Q}
    end.
```
Good find!